src/lib/dnssd/minimal_mdns/core/RecordWriter.cpp

```diff
@@ -41,7 +41,7 @@ RecordWriter & RecordWriter::WriteQName(const FullQName & qname)
     size_t qNameWriteStart = mOutput->WritePos();
     bool isFullyCompressed = true;
 
-    for (uint16_t i = 0; i < qname.nameCount; i++)
+    for (uint16_t i = 0; i < static_cast<uint16_t>(qname.nameCount); i++)
```

Comment:
So we do the following with `i`:

1. Add it to the `names` pointer.
2. Add it to `nameCount`, which is size_t.
3. Use it to index into `names` (which is again adding it to pointer).

Is there a reason `i` is not just `size_t`?

Reply:
Updated.